app/controllers/readings_controller.rb

```diff
@@ -1,10 +1,33 @@
+# require 'time'
+
 class ReadingsController < ApplicationController
   def index
+    @readings = Reading.where(sensor_id: params[:sensor_id])
+    @sensor = Sensor.find(params[:sensor_id])
+    @user = User.find(params[:user_id])
   end
 
   def show
+    @reading = Reading.find(params[:id])
   end
 
   def new
+    @reading = Reading.new
+    @sensor = Sensor.find(params[:sensor_id])
+    @user = User.find(params[:user_id])
```

Comment:
if you declare a `user` to pass it to the view, you should consider creating a helper method in the `readings_helper.rb`, that will give you the user information, something like `current_user`